### ISSUE TITLE
🐛 Bug Report: ADR Fails to Load ADRS from Azure Dev Ops

### ISSUE THREAD

**Issue by JarrodJ83** [backstage/backstage#22102]
### 📜 Description

When using Azure Dev Ops w/ the ADR plugin it fails to load the details of the ADRs with the following error: 

```
Failed to fetch ADR

NotFoundError: https://dev.azure.com/org/project/_git/myrepo?path=%2Fdocs%2Fadrs&version=GBmain/0001-myadr.md could not be read as https://dev.azure.com/org/project/_apis/git/repositories/myrepo/items?api-version=6.0&path=%2Fdocs%2Fadrs&version=main%2F0001-myadr.md, 404 Not Found
```

The issue appears to be a malformed url to the ADR. It seems the plugin assumes the `path` is the last query param returned by ADO and appends the ADR's path to the url. However, ADO is actually returning `&path=%2Fdocs%2Fadrs&version=main%2F` in the url so when the plugin appends the adr's path it is not actually applied to the `path` query param. Full conversation about this available in Discord: https://discord.com/channels/687207715902193673/1185009173050695700

### 👍 Expected behavior

ADR loads and displays correctly

### 👎 Actual Behavior with Screenshots

![image](https://github.com/backstage/backstage/assets/248166/d8eb3cc0-2e36-4d0a-b14b-f7800bf704f8)

### 👟 Reproduction steps

Use the plugin with Azure Dev Ops

### 📃 Provide the context for the Bug.

Issue renders the plugin unusable with ADO.

Additional context: https://discord.com/channels/687207715902193673/1185009173050695700

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by kuangp** [comment]
Thanks for reporting this issue! This looks like an oversight due to assumptions made when this was originally implemented to work with GitHub only but it seems ADO urls work a bit different. 

I think the solution here might be to create an additional util method similar to [`getAdrLocationUrl()`](https://github.com/backstage/backstage/blob/master/plugins/adr-common/src/index.ts#L57-L69), maybe called something like `getAdrFileLocationUrl()` that leverages the `ScmIntegrationRegistry` to resolve the correct url format according to the SCM system being used:

```ts
export const getAdrFileLocationUrl = (
  entity: Entity,
  adrFilePath: string,
  scmIntegration: ScmIntegrationRegistry,
) => {
  if (!isAdrAvailable(entity)) {
    throw new Error(`Missing ADR annotation: ${ANNOTATION_ADR_LOCATION}`);
  }

  return scmIntegration.resolveUrl({
    url: `${getAdrLocationDir(entity)!.replace(/\/$/, '')}/${adrFilePath}`,
    base: getEntitySourceLocation(entity).target,
  });
};
```

and then update the code [here](https://github.com/backstage/backstage/blob/5ccf26657140f7fd141e0a483da5b001576c6e26/plugins/adr/src/components/AdrReader/AdrReader.tsx#L47-L49) to the following:

```diff
- const adrLocationUrl = getAdrLocationUrl(entity, scmIntegrations);

- const url = `${adrLocationUrl.replace(/\/$/, '')}/${adr}`;
+ const url = getAdrFileLocationUrl(entity, adr, scmIntegration);
```

I don't have an ADO instance available and haven't tested the above but if someone were willing to try it out to confirm if that fixes the issue and open a PR that would be much appreciated 🙏

**Comment by JarrodJ83** [comment]
I can give it a shot. I will let you know how it goes.

**Comment by JarrodJ83** [comment]
I tried to get this working locally but when I go to the ADR tab it fails to load with `Error: ENOTEMPTY: directory not empty, rmdir 'C:\Users\jjohnson\AppData\Local\Temp\backstage-tmprktx0U'`. It did this even before I made the code changes and setup azure integration. Any idea what would cause it?

**Comment by JarrodJ83** [comment]
@kuangp I'd really like to help test your changes but I have not been able to get passed the above issue that is preventing me from testing the code. Any ideas?

**Comment by deepan10** [comment]
Hello, 
I replicated this issue in Azure DevOps and the suggestion https://github.com/backstage/backstage/issues/22102#issuecomment-1877272355 is not working. since it's still appending the ADR file path at last `https://dev.azure.com/org/project/_git/myrepo?path=%2Fdocs%2Fadrs&version=GBmain/0001-myadr.md`. I kinda found the solution but doing some more digging on this issue and will raise the PR.

**Comment by deepan10** [comment]
The [Azure DevOps Discovery](https://backstage.io/docs/integrations/azure/discovery) offers to fetch the catalog from different branches, if the Adopter uses the branch the target location URL have the parameter versions point the desired branch name. When it come to ADR plugin the ADR `file` backend call append the ADR file at end of the URL after SCM Integration Resolver. 

Many cases the version is placed in the 1st parameter `https://dev.azure.com/org/project/_git/myrepo?version=GBmain&path=%2F`. 

If the version placed in 2nd parameter like `https://dev.azure.com/org/project/_git/myrepo?path=%2F&version=GBmain`. Then the ADR Backend `file` call append the file name at the end. like `https://dev.azure.com/org/project/_git/myrepo?path=%2F&version=GBmain/0001-my-adr.md` and it was failing. PR #23007 solves this issue with our interrupt the other SCM tools.

@kuangp hope you can review the PR and comment.

**Comment by Rugvip** [comment]
Closing as fixed by #23007, thank you @deepan10! 👍
